.github/workflows/tests.yml

```diff
@@ -8,13 +11,59 @@ on:
   workflow_dispatch:
 
 jobs:
-  unittest-and-lint:
-    runs-on: ${{matrix.os}}
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+
+      - name: Setup up Linter
+        run: |
+          pip install toml yapf
+
+      - name: Lint code formatting
+        # make yapf fail if any file needs to be formatted
+        run: |
+          yapf -r --diff examples/ odxtools/ tests/
+
+  static-type-checking:
+    runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        # restrict the matrix to the oldest and the latest Python
+        # version being supported by odxtools
+        python-version: ["3.8", "3.11"]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{matrix.python-version}}
+          cache: "pip"
+
+      - name: Install odxtools development and testing dependencies
+        run: |
+          pip install -r requirements.txt
+          pip install -r test-requirements.txt
```

Comment:
Are those requirements really needed to just run mypy ?

Reply:
Except for `coverage`, which is contained in `test-requirements.txt`, mypy needs at least stubs with type annotations of every imported package.